streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -188,7 +188,8 @@ private SourceNodeFactory(String name, String[] topics, Pattern pattern, Deseria
         @SuppressWarnings("unchecked")
         @Override
         public ProcessorNode build() {
-            return new SourceNode(name, nodeToSourceTopics.get(name), keyDeserializer, valDeserializer);
+            String[] sourceTopics = nodeToSourceTopics.get(name);
```

Comment:
I know this is a bit opinionated, but ... I think we should make an effort to make all locals `final` where possible. It is just a few extra keystrokes (that intellij can do for you!), and it helps to eliminate a class of bugs.